### PR TITLE
Vela APIServer and Catalog Design

### PR DESCRIPTION
**Pull request by hongchaodeng**
This PR provides a top-down architecture design for Vela APIServer and Catalog. It clarifies the API interfaces for platform builders to build integration solutions, and describes the architecture design in details for incoming roadmap.

### CODE REVIEW

**Comment by linfan** [comment]
The new Catalog parse and search API is highly welcomed : )

**Comment by linfan** [comment]
Is any timeline or milestones available ?

**Comment by hongchaodeng** [comment]
Hi @linfan

I think for the first step we will implement the catalog API and format first. Currently there is no roadmap doc here. I am working on that right now and will follow up in another PR.

File: design/vela-core/APIServer-Catalog.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+
+Here is what a full platform deployment would look like:
+
+![alt](../../docs/resources/api-workflow.png)
+
+### 2. API Design
+
+Below is the overall architecture of API grouping and storage:
+
+![alt](../../docs/resources/api-arch.jpg)
```
Comment (zzxwill) [inline design/vela-core/APIServer-Catalog.md:44]: These image links seem broken.

**Review by zzxwill — APPROVED** [review]
LGTM

**Review by wonderflow — APPROVED** [review]
LGTM

**Comment by wonderflow** [comment]
please rebase, the CI was fixed in master branch

**Comment by baxiaoshi** [comment]
I have a few questions about this design：
1.In the design, an application can only be long to one environment. I can't understand why it is designed in this way. In a company, an application is usually deployed in multiple environments
2.The current data storage is not all stored in MySQL. When will it be implemented
3.How to describe how to place multiple applications in multiple clusters in workload

**Comment by hongchaodeng** [comment]
> 1.In the design, an application can only be long to one environment. I can't understand why it is designed in this way. In a company, an application is usually deployed in multiple environments

Could you provide an example?
In current design, environments are actually used to isolate applications and their dependencies. For example, staging and production.

> 2.The current data storage is not all stored in MySQL. When will it be implemented

What's the question?

> 3.How to describe how to place multiple applications in multiple clusters in workload

This is supported in the design.
Can you create an issue? I will provide more details to cover it in the doc.

**Pull request merged**